src/torrent_info.cpp

```diff
@@ -571,14 +571,24 @@ namespace {
 
 	bool extract_files2(bdecode_node const& tree, file_storage& target
 		, std::string const& root_dir, ptrdiff_t const info_ptr_diff
-		, bool const top_level, error_code& ec)
+		, bool const has_files, int const depth, error_code& ec)
```

Comment:
I renamed `top_level` and inverted its meaning, to avoid colliding with the local variable `single_file`.